packages/atomic/src/components/search/atomic-result-link/atomic-result-link.spec.ts

```diff
@@ -0,0 +1,348 @@
+      const element = new AtomicResultLink();
+      const superSpy = vi.spyOn(
+        Object.getPrototypeOf(Object.getPrototypeOf(element)),
+        'connectedCallback'
+      );
+      element.connectedCallback();
+      expect(superSpy).toHaveBeenCalled();
+      superSpy.mockRestore();
+    });
+  });
+
+  describe('#disconnectedCallback', () => {
+    it('should call cleanup function when component is disconnected', async () => {
+      const cleanupSpy = vi.fn();
+      const {element} = await renderComponent();
+
+      // @ts-expect-error private property access for testing
+      element.removeLinkEventHandlers = cleanupSpy;
+
+      element?.disconnectedCallback();
+
+      expect(cleanupSpy).toHaveBeenCalledOnce();
+
+      // @ts-expect-error private property access for testing
+      expect(element.removeLinkEventHandlers).toBeUndefined();
+    });
+  });
+
+  describe('#render', () => {
+    it('should use result clickUri as href when hrefTemplate is not provided', async () => {
+      const {link} = await renderComponent({
+        result: buildFakeResult({clickUri: 'https://example.com/click'}),
+      });
+
+      await expect
+        .element(link)
+        .toHaveAttribute('href', 'https://example.com/click');
+    });
+
+    it('should build href from template using result properties', async () => {
+      const templateString =
+        '$' + '{clickUri}' + '?' + 'id=' + '$' + '{raw.urihash}';
+      const {link} = await renderComponent({
+        props: {hrefTemplate: templateString},
+        result: buildFakeResult({
+          clickUri: 'https://example.com/test',
+          raw: {urihash: 'hash123'},
+        }),
+      });
+
+      await expect
+        .element(link)
+        .toHaveAttribute('href', 'https://example.com/test?id=hash123');
+    });
+
+    it('should handle complex template with multiple substitutions', async () => {
+      const {link} = await renderComponent({
+        props: {
+          hrefTemplate: '$' + '{clickUri}' + '?title=' + '$' + '{title}',
+        },
+        result: buildFakeResult({
+          clickUri: 'https://example.com/test',
+          title: 'Test Title',
+          uri: 'https://example.com/original',
+        }),
+      });
+
+      await expect
+        .element(link)
+        .toHaveAttribute('href', 'https://example.com/test?title=Test Title');
+    });
+
+    it('should render atomic-result-text with title field when hasDefaultSlot is false', async () => {
+      const {element} = await renderComponent();
+
+      await new Promise((resolve) => setTimeout(resolve, 10));
+      const atomicResultText = element.querySelector('atomic-result-text');
+      expect(atomicResultText).toBeDefined();
+
+      if (atomicResultText) {
+        expect(atomicResultText.getAttribute('field')).toBe('title');
+        expect(atomicResultText.getAttribute('default')).toBe('no-title');
+      }
+    });
+
+    it('should render slotted content when hasDefaultSlot is true', async () => {
```

Comment:
The component no longer has a `hasDefaultSlot` attribute. Also I'm not a fan of the setTimeout if we can avoid it (we can probably use fake timers instead).